src/Dialect/ONNX/ONNXOps/ControlFlow/Loop.cpp

```diff
@@ -30,8 +30,13 @@ using namespace onnx_mlir;
 LogicalResult ONNXLoopOp::inferShapes(
     std::function<void(Region &)> doShapeInference) {
   auto builder = Builder(getContext());
+
+  size_t numCarried = getVInitial().size();
+
   auto &loopBody = getRegion();
-  assert(loopBody.getNumArguments() >= 2 &&
+  // Body inputs: trip count, termination condition, loop carried dependencies.
+  // TODO: Add verifier to check this.
+  assert(loopBody.getNumArguments() == 2 + numCarried &&
          "Loop body must take at least 2 inputs.");
```

Comment:
update the assertion failure message according to new condition

Reply:
changed to "LoopOp inputs count must match body operands count"

also added a message to the corresponding assert in Scan.cpp